middleware/error-handler.js

```diff
@@ -1,10 +1,37 @@
-const { CustomAPIError } = require('../errors')
-const { StatusCodes } = require('http-status-codes')
+// const { CustomAPIError } = require('../errors');
+const { StatusCodes } = require('http-status-codes');
 const errorHandlerMiddleware = (err, req, res, next) => {
-  if (err instanceof CustomAPIError) {
-    return res.status(err.statusCode).json({ msg: err.message })
+  let customError = {
+    //set default
+    statusCode: err.statusCode || StatusCodes.INTERNAL_SERVER_ERROR,
+    msg: err.message || 'Something went wrong try again later',
+  };
+
+  // if (err instanceof CustomAPIError) {
+  //   return res.status(err.statusCode).json({ msg: err.message });
+  // }
+
+  if (err.name === 'ValidationError') {
+    customError.msg = Object.values(err.errors)
+      .map((item) => item.message)
+      .join(',');
+    customError.statusCode = 400;
+  }
+
+  if (err.name === 'CastError') {
+    customError.msg = `No item found with id: ${err.value}`;
+    customError.statusCode = 404;
+  }
+  //if there is error object check if that err.code is === 1100
+  if (err.code && err.code === 11000) {
```

Comment:
Avoid "magic numbers," in your code. What is 11000? In this case, there's a reasonable amount of context where we could make a guess, but generally always give magic numbers a name. For example, you can put this at the top of the file:

```
const MONGO_ERR_DUPLICATE_ENTRY = 11000
```

Again, the point is to give this number a name!